File: safe_crash.cpp
/*
------------------ TCS MOCKVITA ROUND 1---------------------
*/
#include <bits/stdc++.h>

using namespace std;

#define fastio ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

typedef long long ll;
typedef unsigned long long ull;
typedef unsigned int ui;
typedef unsigned short us;
typedef long l;

int main(){
	fastio;
	ll c;
	cin>>c;
	map<ull,ull> time;
	while (c--){
		ll a,b;
		ull v;
		cin>>a>>b>>v;
		time[(ull)sqrt(a*a+b*b)/v]++;
	}
	//cout<<"cp1\n";
	map<ull,ull>::iterator itr;
	ull sum=0;
	for (itr=time.begin(); itr!=time.end(); ++itr){
		sum+=(itr->second)*(itr->second-1)/2;
	} 
	cout<<sum<<endl;
	return 0;
}

File: safe_crash.py
'''
--------------------TCS MOCKVITA ROUND1----------------------

'''

from collections import Counter
from math import sqrt

n = int(input())
time = []
for _ in range(n):
    x,y,v = map(int,input().strip().split())
    time.append(sqrt(x**2+y**2)/v)
time = Counter(time)
print(sum([c*(c-1)//2 for c in time.values()]))